src/components/FilterBadges/FilterBadges.stories.tsx

```diff
@@ -1,49 +1,76 @@
-import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
-import { ProjectFilterOptions } from "types/commits";
-import { queryString, url } from "utils";
-import { FilterBadges } from ".";
-
-const { upsertQueryParam } = url;
-const { parseQueryString } = queryString;
+import { useState } from "react";
+import Button from "@leafygreen-ui/button";
+import TextInput from "@leafygreen-ui/text-input";
+import FilterBadges from ".";
+import { FilterBadgeType } from "./FilterBadge";
 
 export default {
   title: "FilterBadges",
-  decorators: [
-    (Story) => (
-      <BrowserRouter>
-        <Story />
-      </BrowserRouter>
-    ),
-    withKnobs,
-    withQuery,
-  ],
   component: FilterBadges,
 };
+
 export const Default = () => {
-  const updateQueryParams = useUpdateURLQueryParams();
-  const { search } = useLocation();
-  const queryParams = parseQueryString(search);
+  const [badges, setBadges] = useState<FilterBadgeType[]>([
+    { key: "test", value: "test" },
+  ]);
 
-  const badgeKey = text("Badge Key", "");
-  const badgeValue = text("Badge Value", "");
-  const addBadge = () => {
-    const params = upsertQueryParam(queryParams[badgeKey], badgeValue);
-    updateQueryParams({ [badgeKey]: params });
+  const addBadge = (key: string, value: string) => {
+    setBadges([...badges, { key, value }]);
+  };
+  const removeBadge = (badge: FilterBadgeType) => {
+    setBadges(
+      badges.filter(
+        (b) => (b.key !== badge.key && b.value !== badge.value) || false
+      )
+    );
   };
-  button("Add Badge", addBadge);
+  const onClearAll = () => {
+    setBadges([]);
+  };
+
   return (
-    <FilterBadges
-      queryParamsToDisplay={new Set([ProjectFilterOptions.BuildVariant])}
-    />
+    <div>
+      <FilterBadges
+        badges={badges}
+        onRemove={removeBadge}
+        onClearAll={onClearAll}
+      />
+      <BadgeInput onAdd={addBadge} />
+    </div>
   );
 };
 
-Default.parameters = {
-  query: {
-    buildVariants:
-      "! Enterprise Clang Tidy,! Enterprise Windows,Enterprise RHEL 8.0 (Lock Free Reads disabled)",
-  },
+// Must use a seperate input component to dynamically add badges
+// Since leafygreen knobs rerender the component on every change
+const BadgeInput = ({
+  onAdd,
+}: {
+  onAdd: (key: string, value: string) => void;
+}) => {
+  const [badgeKey, setBadgeKey] = useState("");
+  const [badgeValue, setBadgeValue] = useState("");
+
+  const handleAdd = () => {
+    onAdd(badgeKey, badgeValue);
+    setBadgeKey("");
+    setBadgeValue("");
+  };
+  return (
+    <div>
+      <TextInput
+        placeholder="key"
+        label="key"
+        value={badgeKey}
+        onChange={(e) => setBadgeKey(e.target.value)}
+      />
+      <TextInput
+        placeholder="value"
+        label="value"
+        value={badgeValue}
+        onChange={(e) => setBadgeValue(e.target.value)}
+        onSubmit={handleAdd}
+      />
+      <Button onClick={() => onAdd(badgeKey, badgeValue)}>Add</Button>
```

Comment:
This seems a little strange to me

Maybe `L71` should be deleted, and `L73` should say `onClick={() => handleAdd()}`?